### PR TITLE
ci(retest): filter more comments to use GHA less

### PR DESCRIPTION
**Pull request by agilgur5**
Follow-up to #13000

### Motivation

- they get triggerred [quite often](https://github.com/argoproj/argo-workflows/actions/workflows/retest.yaml) (for every comment), and though they only run for 2s, it would be good to filter out more of them and use less CI time

### Modifications

- only run on `created` events, skip all `edited` events
  - the vast majority of `edited` events will be for unrelated comments
  - you can delete and comment again to trigger a retest instead if you had a typo

- only run on PR comments, following [GH docs](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#issue_comment-on-issues-only-or-pull-requests-only)
  - it would fail otherwise, but this prevents that scenario entirely

### CODE REVIEW

**Review by juliev0 — APPROVED** [review]

File: .github/workflows/retest.yaml [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
+    types: [created]
 
 permissions:
   contents: read
 
 jobs:
   retest:
-    if: github.event.comment.author_association == 'MEMBER' && github.event.comment.body == '/retest'
+    # PR comments where a Member types "/retest" exactly
+    if: github.event.issue.pull_request && github.event.comment.author_association == 'MEMBER' && github.event.comment.body == '/retest'
```
Comment (juliev0) [inline .github/workflows/retest.yaml:12]: So, this only works for Members? Can I assume that Reviewers, Approvers, and Leads are all in the 'MEMBER' author_association?

Comment (agilgur5) [reply .github/workflows/retest.yaml:12]: Yes, it's anyone in the Argoproj org. Note that we all also have the "Member" next to our name in comments.

Also that part hasn't changed in the PR 😅

Comment (juliev0) [reply .github/workflows/retest.yaml:12]: > Note that we all also have the "Member" next to our name in comments.
Got it!

> Also that part hasn't changed in the PR
Hey, I can still ask questions :)

Comment (agilgur5) [reply .github/workflows/retest.yaml:12]: For sure! I think I missed that you approved first and then commented

**Comment by agilgur5** [comment]
> Verification

Can confirm this worked in https://github.com/argoproj/argo-workflows/pull/13317#issuecomment-2211986079

**Pull request merged**